README.md
# nagios
Nagios nice scripts

based on

http://www.unixmen.com/how-to-install-nagios-4-1-in-ubuntu-15-04/
https://www.digitalocean.com/community/tutorials/how-to-install-nagios-4-and-monitor-your-servers-on-ubuntu-14-04

http://www.techienote.com/pnp4nagios-on-ubuntu/

https://docs.pnp4nagios.org/pnp-0.6/verify_pnp_config

http://dev.vemarsas.it/documents/29


nagios-plugins-2.0.3/ChangeLog
2014-06-26  abrist  <abrist@nagios.com>

	* NP-VERSION-GEN, configure.ac: Prep for 2.0.3

	* THANKS.in: Updated THANKS.in

	* NEWS: Updated NEWS for 2.0.3

2014-06-26  Spenser Reinhardt  <sreinhardt@nagios.com>

	* gl/getopt.c, lib/utils_cmd.c: whitespace cleanup

2014-06-23  Spenser Reinhardt  <sreinhardt@nagios.com>

	* lib/utils_cmd.c: lib/utils_cmd.c - Free file descriptor

	Coverity 66502 - File descriptor fd in cmd_file_read is never closed, and thus file is left open after usage throughout runtime. - SR

2014-06-22  Spenser Reinhardt  <sreinhardt@nagios.com>

	* plugins/negate.c: plugins/negate.c - Function should not return.

	Coverity 66479 - validate_arguments has no need to return anything, as it dies on error, yet was set to return an int. Set to void to resolve warning.

	* plugins/negate.c: plugins/negate.c - Reorder if statement, aiob

	Coverity 66480 - Potential array index out of bounds, since result was not verified to be positive prior to using as an index for state[]. Simply reording the if statement should resolve the issue. - SR

	* plugins-root/check_dhcp.c: plugins-root/check_dhcp.c - array out of
	bounds

	Coverity 66488 - offer_packet->options has a max size of 312. It was being used in a loop verifying less than 311, but increasing by 2 per loop, causing a possible array index out of bounds. Changed to checking less than max length - 1. - SR

	* plugins/check_http.c: plugins/check_http.c - leakage fix

	Coverity 66514 - Possible leakage and overflow with addr in redirect functionality. Not confirmed as null terminated, and externally gathered. Restrict string comparisons and duplications by size. - SR

	* plugins/check_ntp.c: plugins/check_ntp.c - Verify struct from response

	Coverity 66524 - req.data is not neccessarily null terminated but still feed to printf statements. This both does that, and verifies the struct more so than before. - SR

2014-06-19  Spenser Reinhardt  <sreinhardt@nagios.com>

	* lib/parse_ini.c: lib/parse_ini.c - Cleaner and more flexible - suid

	Adapted from M-P 91d04ad6 and 48025ff3 around the existing changes.

	* lib/parse_ini.c: lib/parse_ini.c - Initial drop privs

	Handling the race condition left from suid bins. Moving previous patches from my private copy.

	* gl/Makefile.am, gl/m4/gnulib-cache.m4: Gnulib module "idpriv-droptemp"
	finishing touches

2014-06-18  Holger Weiss  <holger@zedat.fu-berlin.de>

	* gl/idpriv-droptemp.c, gl/idpriv.h, gl/m4/gnulib-comp.m4,
	gl/m4/idpriv.m4: Add Gnulib module "idpriv-droptemp"

2014-06-05  Spenser Reinhardt  <sreinhardt@nagios.com>

	* plugins/check_real.c: plugins/check_real.c - recv string null
	terminate

	Recv into buffer is not properly null terminated prior to strstr and possible other string functions expecting a null termination. Simply take bytes received and use as an index to append \0 after. We are creating buffer[] with size of MAX_INPUT_BUFFER and recv with MAX_INPUT_BUFFER-1 so this should never overflow.

	* plugins/check_apt.c: plugins/check_apt.c - Print uninitialized ereg

	Coverity 66531 - ereg.buffer can be printed without being initialized if do_include and do_exclude are null and critical is an invalid regex. While minor this may leak memory and cause undefined behavior.

2014-06-05  Eric J. Mislivec  <emislivec@nagios.com>

	* plugins/check_tcp.c, plugins/negate.c: Include common.h before any
	system headers.

	This should fix some problems building on AIX.

2014-06-03  Spenser Reinhardt  <sreinhardt@nagios.com>

	* plugins/check_dns.c: check_dns.c - Fixing querytypes ALL to ANY

	Mislabeled nslookups ANY type to ALL.

	* plugins/check_dns.c: check_dns.c - Reverse dns lookup fix

	Query type changes caused a bug with reverse lookup matching a hostname, not actually matching.

2014-06-03  abrist  <abrist@nagios.com>

	* plugins-scripts/check_mailq.pl: Revert "check_mailq: try to autodetect
	which mailq implementation we are using"

	This reverts commit bc4113fba2d251b70f8d9dae80f1b18033cace40.

	* plugins-scripts/check_mailq.pl: Revert "check_mailq: document
	autodetection in the usage output"

	This reverts commit 8e087cf4a6f741e35649d5df3e9b3a57615ff0a9.

	* plugins-scripts/check_mailq.pl: Revert "check_mailq: add $mailq to
	check output, so it is easily visible what was autodetected"

	This reverts commit 21750b173b8d98eaddcf99b7372236a2df2eb4c4.

2014-05-20  abrist  <abrist@nagios.com>

	* NP-VERSION-GEN, configure.ac: Prep for 2.0.2

	* NEWS, THANKS.in: Updated NEWS and THANKS.in

2014-04-21  Gunnar Beutner  <gunnar@beutner.name>

	* configure.ac, plugins/check_ping.c: Make check_ping work on Windows.

	* configure.ac: Add missing file extensions for some of the plugins.

	* plugins/check_disk.c: Make check_disk work on Windows.

	* plugins/Makefile.am, plugins/check_users.c: Make check_users work on
	Windows.

2014-05-19  Spenser Reinhardt  <sreinhardt@nagios.com>

	* lib/parse_ini.c: File access vulnerability with SUID bins

	Check_dhcp and check_icmp are root setuid binaries, with extra-opts enabled, file permissions are never checked before fopen is called on files. This patch uses POSIX standard functions, so they should be portable, to verify real uid should have permissions to config file and that it is also not a linked file. These changes are global and WILL effect non-suid plugins as well, however considering real and effective uid should be the same in non-suid applications, it should just properly verify file permissions before attempting to access and not cause issues provided file access is permitted. - SR

	http://www.exploit-db.com/exploits/33387/

	Thanks to Dawid Golunski for the submission, and emislivec for reworking the patch a bit.

2014-05-19  abrist  <abrist@nagios.com>

	* plugins/netutils.c: netutils.c - A few more changes

	Changed bool to short.

	Removed first instance of is_socket to avoid
	redeclaration error.

	Changed 'socket' to 'file socket' for verbosity.

2014-05-07  Davide Madrisan  <davide.madrisan@gmail.com>


install_nagios_4_ubuntu_new.sh
    sudo apt-get install build-essential libgd2-xpm-dev apache2-utils unzip
    sudo useradd -m nagios
    sudo passwd nagios
    9  sudo groupadd nagcmd
   10  sudo usermod -a -G nagcmd nagios
   11  sudo usermod -a -G nagcmd www-data
   12  wget https://assets.nagios.com/downloads/nagioscore/releases/nagios-4.1.0rc2.tar.gz
   13  wget http://nagios-plugins.org/download/nagios-plugins-2.0.3.tar.gz
   14  tar xzf nagios-4.1.0rc2.tar.gz
   15  cd nagios-4.1.0rc2/
   16  sudo ./configure --with-command-group=nagcmd
   17  sudo make all
   18  sudo make install
   19  sudo make install-init
   20  sudo make install-config
   21  sudo make install-commandmode
   22  sudo make install-webconf
   23  ll
   24  sudo /usr/bin/install -c -m 644 sample-config/httpd.conf /etc/apache2/sites-enabled/nagios.conf
   25  apt-get install apache2 -y
   26  sudo /usr/bin/install -c -m 644 sample-config/httpd.conf /etc/apache2/sites-enabled/nagios.conf
   27  sudo ls -l /etc/apache2/sites-enabled/
   28  sudo htpasswd -c /usr/local/nagios/etc/htpasswd.users nagiosadmin
   29  sudo htpasswd -c /usr/local/nagios/etc/htpasswd.users nagios
   30  sudo htpasswd -c /usr/local/nagios/etc/htpasswd.users nagiosadmin
   31  sudo systemctl restart apache2
   32  sudo service apache2 restart
   33  cd ..
   34  tar xzf nagios-plugins-2.0.3.tar.gz
   35  cd nagios-plugins-2.0.3/
   36  sudo ./configure --with-nagios-user=nagios --with-nagios-group=nagios
   37  sudo make
   38  sudo make install
   39  sudo a2enmod rewrite
   40  ubuntu-amd64:/home/ubuntu/nagios-plugins-2.0.3#sudo a2enmod cgi
   41  sudo a2enmod cgi
   42  sudo service apache2 restart
   43  sudo service nagios start
   44  sudo ln -s /etc/init.d/nagios /etc/rcS.d/S99nagios
   45  ifconfig 
   46  reboot
   47  vi nagios.conf 
   48  sudo service apache2 restart
   49  vi nagios.conf 
   50  apt-get install php5
   51  sudo service apache2 restart
   52  cd
   53  wget https://mathias-kettner.de/download/check_mk-1.2.7i2p2.tar.gz
   54  ll
   55  tar xvfz check_mk-1.2.7i2p2.tar.gz 
   56  cd check_mk-1.2.7i2p2
   57  ll
   58  tar xvfz agents.tar.gz 
   59  ll
   60  ./setup.sh 
   61  vi 
   62  vim /root/.check_mk_setup.conf 
   63  ./setup.sh 
   64  cd /etc/check_mk/
   65  ll
   66  find ./
   67  cat main.mk
   68  vi main.mk
   69  cmk -Ivp
   70  cd /root/
   71  ll
   72  cd check_mk-1.2.7i2p2
   73  ll
   74  dpkg -i check-mk-agent_1.2.7i2p2-1_all.deb
   75  apt-get install xinetd
   76  dpkg -i check-mk-agent_1.2.7i2p2-1_all.deb
   77  cmk -Ivp
   78  cmk -Rvp
   79  cd /etc/check_mk/
   80  ll
   81  vi main.mk
   82  cmk -IIvp
   83  cmk -Ivp
   84  vi main.mk
   85  cmk -Ivp
   86  cmk -Rvp
   87  vi main.mk
   88  cmk -Ivp
   89  cmk -Rvp
   90  cd /etc/nanorc 
   91  cd /usr/local/nagios/
   92  ll
   93  cat /root/.check_mk_setup.conf 
   94  vim /root/.check_mk_setup.conf 
   95  cd /usr/local/nagios/etc/
   96  ll
   97  grep -R localhost
   98  mv objects/localhost.cfg objects/localhost.cfg.bak
   99  cmk -Rvp
  100  service nagios restart
  101  mv objects/localhost.cfg.bak objects/localhost.cfg
  102  vi /etc/hosts
  103  vi /etc/check_mk/main.mk
  104  cmk -IIvp
  105  cmk -Rvp
  106  ll
  107  cd objects/
  108  ll
  109  cat localhost.cfg 
  110  vi /etc/check_mk/main.mk
  111  cmk -IIvp
  112  cmk -Rvp
  113  vi /etc/check_mk/main.mk
  114  cmk -IIvp
  115  cmk -Rvp
  116  service nagios restart
  117  reboot
  118  ll
  119  cd /usr/local/nagios/etc/
  120  ll
  121  cat nagios.cfg 
  122  grep log
  123  grep log *
  124  cat /usr/local/nagios/var/nagios.log
  125  ll /usr/lib/check_mk/livestatus.o
  126  vi /usr/lib/check_mk/livestatus.o
  127  ldd /usr/lib/check_mk/livestatus.o
  128  apt-get install -f
  129  apt-get install linux-vdso
  130  rm /usr/lib/check_mk/livestatus.o
  131  cmk -Rvp
  132  cat /usr/local/nagios/var/nagios.log
  133  w
  134  history
